RocketChat/Rocket.Chat.ReactNative
# Chore: Update react-native-mmkv-storage to 0.6.12

## Proposed changes
The proposal of this PR is update the library `react-native-mmkv-storage` to the latest version (0.6.12), to have access to JSI support, and with that, change all mmkv calls from `asynchronously` to `synchronously`.
Consequently, we will have a performance gain because it doesn't need to cross the bridge anymore.

## Issue(s)	
N/A

## How to test or reproduce
As this PR changed all mmkv calls, will be necessary test all features related about it.

- At workspace screen:
  - Test input a new workspace;
  - Test the "join our open workspace" button;
  - Test the login with a certificate; 
- At home screen:
  - Test the "Add server" functionality;
  - with more then one server, test change the servers;
- At home screen -> open the menu -> settings -> theme: 
  - Change the theme, close the app and verify if when open the app again, the selected theme is still select it;
- At home screen, enter your E2E password from the top option:
  - go to menu -> settings -> Security and privacy -> E2E Encyption: 
  - test the rest button and verify if is still working; 
-  At home screen -> open the menu -> settings -> Security and privacy -> Screen lock, then test if this functionality is still working;
-  At home screen -> open the menu -> settings -> Default browser
   - After that, close and open the app, go to some chat and try open a link and verify if will open at the selected browser;
- try share a image or document from app gallery;
- Share extension

Other importants tests: 

- SSL client certificate is working (already descibed above, at login flow);
- Enable "FileUpload_ProtectFiles" setting and make sure push notifications are showing avatars;
- Enable E2E encryption and make sure push notifications are showing the message content;
- Enable "Push_request_content_from_server" and make sure push notifications are showing content;
- Make sure it's backwards compatible to the previous version of the app:
  - Install the current version of the app, update to this one and all keys need to be there (e.g. the user remains logged in);


## Screenshots
N/A

## Types of changes




## Checklist



## Further comments
After reviewed and tested, will be necessary to merge the changes from this PR: https://github.com/RocketChat/react-native/pull/1
